src/ring_theory/filtration.lean

```diff
@@ -7,13 +7,32 @@ Authors: Andrew Yang
 import ring_theory.noetherian
 import ring_theory.rees_algebra
 import ring_theory.finiteness
+import data.polynomial.module
+import order.hom.complete_lattice
 
 /-!
 
 # `I`-filtrations of modules
 
 This file contains the definitions and basic results around (stable) `I`-filtrations of modules.
 
+## Main results
+
+- `ideal.filtration`: An `I`-filtration on the module `M` is a sequence of decreasing submodules
+  `N i` such that `I • N ≤ I (i + 1)`. Note that we do not require the filtration to start from `⊤`.
+- `ideal.filtration.stable`: An `I`-filtration is stable if `I • (N i) = N (i + 1)` for large
+  enough `I`.
+- `ideal.filtration.submodule`: The associated module `⨁ Nᵢ` of a filtration, implemented as a
+  submodule of `M[X]`.
+- `ideal.filtration.submodule_fg_iff_stable`: If `F.N i` are all finitely generated, then
+  `F.stable` iff `F.submodule.fg`.
+- `ideal.filtration.stable.of_le`: In a finite module over a noetherian ring,
+  if `F' ≤ F`, then `F.stable → F'.stable`.
+- `ideal.exists_pow_inf_eq_pow_smul`: **Artin-Rees lemma**
+- `ideal.infi_pow_eq_bot_of_local_ring`:
+  **Krull's intersection theorem** for noetherian local rings.
```

Comment:
Please include human readable statements, like you do for the other items in this list.